lib/share_progress/variant_parser.rb

```diff
@@ -0,0 +1,24 @@
+require 'share_progress'
+require 'share_progress/errors'
+require 'share_progress/email_variant'
+require 'share_progress/facebook_variant'
+require 'share_progress/twitter_variant'
+
+module ShareProgress
+  class VariantParser
+
+    def self.parse(hash_to_parse)
+      keys = hash_to_parse.keys.map(&:to_sym)
+      match = nil
+      [FacebookVariant, EmailVariant, TwitterVariant].each do |variant_class|
+        if (keys & variant_class.fields).size > 0 # & is array intersection
```

Comment:
this checks the size of the overlap between the key lists by using the array intersection operator `&`